daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/grpcLedgerClient/GrpcLedgerClient.scala

```diff
@@ -74,17 +71,22 @@ class GrpcLedgerClient(
     queryWithKey(parties, templateId).map(_.map(_._1))
   }
 
-  // Omits the package id on an identifier if contract upgrades are enabled
-  private def toApiIdentifierUpgrades(identifier: Identifier): api.Identifier = {
+  // Omits the package id on an identifier if contract upgrades are enabled unless explicitPackageId is true
+  private def toApiIdentifierUpgrades(
+      identifier: Identifier,
+      explicitPackageId: Boolean,
+  ): api.Identifier = {
     val converted = toApiIdentifier(identifier)
-    if (providePackageId) converted else converted.copy(packageId = "")
+    if (explicitPackageId || !enableContractUpgrading) converted else converted.copy(packageId = "")
   }
 
+  // TODO[SW]: Currently do not support querying with explicit package id, interface for this yet to be determined
+  // See https://github.com/digital-asset/daml/issues/17703
```

Comment:
Do notice this

Reply:
I'm confused by this, doesn't `toApiIdentifierUpgrades` keep the explicit package id if `enableContractUpgrading := false`?

Reply:
Ah right, i mean when upgrades are enabled. There was some discussion about wanting to be able to query with and without explicit package id once we have upgrades. Also note that in 2.9/3.0, `enableContractUpgrading` may be true by default

Reply:
gotcha! thanks for the background :)